src/core/tools/generateImageTool.ts

```diff
@@ -177,13 +179,23 @@ export async function generateImageTool(
 			}
 
 			// Create a temporary OpenRouter handler with minimal options
-			const openRouterHandler = new OpenRouterHandler({} as any)
+			// kilocode_change start
+			const openRouterHandler = openRouterApiKey
+				? new OpenRouterHandler({})
+				: new KilocodeOpenrouterHandler({ kilocodeToken: kiloCodeApiKey })
+			// kilocode_change end
 
 			// Call the generateImage method with the explicit API key and optional input image
 			const result = await openRouterHandler.generateImage(
 				prompt,
 				selectedModel,
-				openRouterApiKey,
+				// kilocode_change start
+				openRouterApiKey ||
+					kiloCodeApiKey ||
+					(() => {
+						throw new Error("Unreachable because of earlier check.")
+					})(),
```

Comment:
lol this a fun pattern I've never seen before!

Reply:
in C# you can just do this:
```c#
blah ?? throw new Exception()
```